generator/candidate_data.py

```diff
@@ -0,0 +1,93 @@
+        self.highschool_city = faker.city()
+        self.matura_date = faker.date_between_dates(date_start=datetime(2017, 1, 1),
+                                                    date_end=recruitment_data.end_date)
+        self.matura_result = random.randrange(100)
+        self.test_result = random.randrange(100)
+        if recruitment_data.degree == "MASTER":
+            self.graduation_date = str(faker.date_between_dates(date_start=self.matura_date,
+                                                                date_end=recruitment_data.end_date))
+            self.college_name = "University of " + faker.word()
+            self.faculty = random.choice(FACULTIES)
+            self.field_of_study = random.choice(MAJORS)
+            self.mode = random.choice(MODE)
+            self.average = random.uniform(2.0, 5.0)
+        else:
+            self.graduation_date = None
+            self.college_name = None
+            self.faculty = None
+            self.field_of_study = None
+            self.mode = None
+            self.average = None
+
+    @classmethod
+    def from_previous_recruitment(cls, other, recruitment_data):
+        """
+        Creates candidate data (next recruitment) from other candidate data
+        :param other: Candidate data for previous recruitment
+        :return: Candidate data for next recruitment
+        """
+        candidate = cls.__new__(cls)
+        candidate.recruitment_data = recruitment_data
+        candidate.pesel = other.pesel
+        candidate.name = other.name
+        candidate.city = other.city
+        candidate.region = other.region
+        candidate.country = other.country
+        candidate.highschool = other.highschool
+        candidate.highschool_city = other.highschool_city
+        candidate.matura_date = other.matura_date
+        candidate.matura_result = other.matura_result
+        candidate.test_result = random.randrange(100)
+        if recruitment_data.degree == "MASTER" and other.recruitment_data.degree == 'BACHELOR':
+            candidate.graduation_date = str(faker.date_between_dates(
+                                        date_start=other.recruitment_data.end_date,
+                                        date_end=recruitment_data.end_date))
+            candidate.college_name = "AGH University"
+            candidate.faculty = other.recruitment_data.faculty
+            candidate.field_of_study = other.recruitment_data.major_name
+            candidate.mode = other.recruitment_data.mode
+            candidate.average = random.uniform(2.0, 5.0)
+        else:
+            candidate.graduation_date = other.graduation_date
+            candidate.college_name = other.college_name
+            candidate.faculty = other.faculty
+            candidate.field_of_study = other.field_of_study
+            candidate.mode = other.mode
+            candidate.average = other.average
+        return candidate
```

Comment:
@jmysliv I think it is better to somehow copy whole **other** into **candidate** so that we get rid of 20 lines of code just to copy attributes, maybe some dictionary that stores common attributes?

Reply:
Maybe to copy whole previous recruitment and overwrite changes?